miner/worker.go

```diff
@@ -239,10 +241,12 @@ func (w *worker) start() {
 func (w *worker) stop() {
 	atomic.StoreInt32(&w.running, 0)
 
-	if istanbul, ok := w.engine.(consensus.Istanbul); ok {
-		err := istanbul.StopValidating()
-		if err != nil {
-			log.Error("Error while calling engine.StopValidating", "err", err)
+	if istanbul, ok := w.engine.(*istanbulBackend.Backend); ok {
+		if istanbul.IsValidating() {
```

Comment:
Why the added condition here?

Reply:
So the [miner.Stop()](https://github.com/celo-org/celo-blockchain/blob/c7bea3217e78c76eb96958ccfd27592f99afaba4/eth/backend.go#L618) function gets called when the user (or the e2e test) actually shuts down the node, in which case istanbul.StopValidating() throws an error here since it's already been called (in the code below)

Reply:
Makes sense, but this is introducing a race condition, since you could still have 2 goroutines calling stop. But  it might be good enough for our purposes if the time between the 2 calls to stop is sufficient.